finalitygadget/finalitygadget.go

```diff
@@ -298,28 +307,28 @@ func (fg *FinalityGadget) QueryLatestFinalizedBlock() (*types.Block, error) {
 
 // This function process blocks indefinitely, starting from the last finalized block.
 func (fg *FinalityGadget) ProcessBlocks(ctx context.Context) error {
-	// Start service at last finalized block
-	err := fg.startService()
-	if err != nil {
-		return fmt.Errorf("%v", err)
-	}
-
 	// Start polling for new blocks at set interval
 	ticker := time.NewTicker(fg.pollInterval)
 	defer ticker.Stop()
+
 	for {
 		select {
 		case <-ctx.Done():
 			return nil
 		case <-ticker.C:
-			block, err := fg.queryBlockByHeight(int64(fg.currHeight + 1))
+			latestFinalizedBlock, err := fg.l2Client.HeaderByNumber(ctx, big.NewInt(ethrpc.FinalizedBlockNumber.Int64()))
```

Comment:
if the op node relies on its own finality gadget to track block finality, doesnt this create circular logic where the chain tip never advances? i.e. finality gadget asks l2 client for latest finalized block, but this is never greater than the last processed block because the l2 client never moves to the next block.

Reply:
Yeah it seems that this circular dependency is necessary

1. finality gadget needs to query the last finalised height in OP node and starts from there
2. OP node needs to query finality gadget to track the newly finalised blocks

As long as we have a good strategy for bootstrapping the finality gadget this should be fine.